docs/docs/cloud/how-tos/copy_threads.md

```diff
@@ -0,0 +1,165 @@
+    ```
+
+=== "Javascript"
+
+    ```js
+    import { Client } from "@langchain/langgraph-sdk";
+
+    const client = new Client({ apiUrl:"<DEPLOYMENT_URL>" });
+    const assistantId = agent;
+    const thread = await client.threads.create();
+    ```
+
+=== "CURL"
+
+    ```bash
+    curl --request POST \
+      --url <DEPLOYMENT_URL>/threads \
+      --header 'Content-Type: application/json' \
+      --data '{
+        "metadata": {}
+      }'
+    ```
+
+### Thread creation
+
+In order to copy a thread - you need to create one first. While it won't throw an error if you try to copy a thread before doing any runs, it doesn't really make sense to copy an empty thread, so in practice you need to create the thread and then execute some runs on it.
+
+Here is how you can create a thread:
+
+=== "Python"
+
+    ```python
+    thread = await client.threads.create()
+    ```
+
+=== "Javascript"
+
+    ```js
+    const thread = await client.threads.create();
+    ```
+
+=== "CURL"
+
+    ```bash
+    curl --request POST \
+      --url <DEPLOYMENT_URL>/threads \
+      --header 'Content-Type: application/json' \
+      --data '{
+        "metadata": {}
+      }'
+    ```
+
+There are a few ways to execute runs on a thread. You can [stream runs](https://langchain-ai.github.io/langgraph/cloud/how-tos/#streaming), execute them in the [background](https://langchain-ai.github.io/langgraph/cloud/how-tos/cloud_examples/background_run/), or even [setup cron jobs](https://langchain-ai.github.io/langgraph/cloud/how-tos/cloud_examples/cron_jobs/) to execute runs on a schedule. 
+
+Whichever way you choose to execute runs on a thread is fine, as long as you ensure that you have properly executed at least one run on your thread before proceeding.
+
+## Copying a thread
+
+After you have created a thread and executed some runs on it, you may wish to copy the thread. This will create a new thread with the same history as the existing thread, and then allow you to continue executing runs.
+
+### Create copy
+
+=== "Python"
+
+    ```python
+    copied_thread = await client.threads.copy(<THREAD_ID>)
+    ```
+
+=== "Javascript"
+
+    ```js
+    let copiedThread = await client.threads.copy(<THREAD_ID>);
+    ```
+
+=== "CURL"
+
+    ```bash
+    curl --request POST --url <DEPLOYMENT_URL>/threads/<THREAD_ID>/copy \
+    --header 'Content-Type: application/json'
+    ```
+
+### Verify copy
+
+We can verify that the history from the prior thread did indeed copy over correctly:
+
+=== "Python"
+
+    ```python
+    def remove_thread_id(d):
+      if 'metadata' in d and 'thread_id' in d['metadata']:
+          del d['metadata']['thread_id']
+      return d
+
+    original_thread_history = list(map(remove_thread_id,await client.threads.get_history(<THREAD_ID>)))
+    copied_thread_history = list(map(remove_thread_id,await client.threads.get_history(copied_thread['thread_id'])))
+
+    if original_thread_history == copied_thread_history:
```

Comment:
(nit) maybe just do assert original == copied? hopefully we will never see "histories are different" :)

Reply:
yeah was going to do, but I did this to ensure exact compatibility with CURL command. If you think important to use assert I can do that and just leave the CURL command a little different.